threedi_schema/application/schema.py

```diff
@@ -207,3 +215,98 @@ def upgrade_spatialite_version(self):
                     copy_models(self.db, work_db, self.declared_models)
                 except IntegrityError as e:
                     raise UpgradeFailedError(e.orig.args[0])
+
+    def convert_to_geopackage(self):
+        """
+        Convert spatialite to geopackage using gdal's ogr2ogr.
+
+        Does nothing if the current database is already a geopackage.
+
+        Raises UpgradeFailedError if the conversion of spatialite to geopackage with ogr2ogr fails.
+        """
+        if self.db.get_engine().dialect.name == "geopackage":
+            return
+        # Check if ogr2ogr
+        result = subprocess.run(
+            "ogr2ogr --version",
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+        ## ogr2ogr is installed; make sure the version is high enough and return if not
+        if result.returncode == 0:
+            # get version
+            version = re.findall(r"\b(\d+\.\d+\.\d+)\b", result.stdout)[0]
+            # trim patch version and convert to float
+            float_version = float(version[0 : version.rfind(".")])
+            if float_version < 3.4:
+                warnings.warn(
+                    f"ogr2ogr 3.4 (part of GDAL) or newer is needed to convert spatialite to geopackage "
```

Comment:
The reason to not raise an Exception here is that that would break 3Di on any system with gdal older than 3.4. And I don't think that is not what we want. I do agree that in general, it's not good practice to just allow this.